src/Microsoft.AspNet.OData/HttpRequestScope.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+
+namespace Microsoft.AspNet.OData
+{
+    internal class HttpRequestScope
```

Comment:
Seems like this should be public?

Reply:
I made it internal on purpose because it's just a piece of plumbing helping me to associate the request with the per-request `IServiceScope`. I'm binding HttpRequest(Message) into the container. Is there a use case for changing the HttpRequest(Message) the per-request container is associated with in the middle of request processing?

Reply:
No, I think the request will no change in the request container, it should be immutable. I've heard feedback that it would be useful for extensibility and that would require public access to developers can use it outside of the OData dll.

Reply:
OK.